### ISSUE TITLE
Could not plan the operation

### ISSUE THREAD

**Issue by ZirgVoice** [wundergraph/cosmo#581]
### Component(s)

router

### Component version

0.69.0

### wgc version

0.43.2

### controlplane version

not using

### router version

0.69.0

## Description
There are 2 services, `todo` and `user`. When extending the `Todo` type in the `user` service, I use the `userID` to look up the `user` for the `todo`. Previously, `id` `todo` was not mandatory, now it must be passed, otherwise there will be an error. I do not use `id` todo in the `user` service, but it can be used in other services, which is why it is present in `@key(fields: "id, userID")` in the `todo` service.

`todo` service:
```gql
type Todo @key(fields: "id, userID") {
    id: ID
    name: String!
    userID: ID!
}
```
`user` service:
```gql
type Todo @key(fields: "id, userID") {
    id: ID
    userID: ID @external
    user: User @requires(fields: "userID")
}
```
> [!WARNING]  
> This is an example of a working scheme.
## Steps to Reproduce
Create 2 services `todo` and `user` with the following types in the schema:
`todo` service:
```gql
type Todo @key(fields: "id, userID") {
    id: ID
    name: String!
    userID: ID!
}
```
`user` service:
```gql
type Todo @key(fields: "userID") {
    userID: ID @external
    user: User @requires(fields: "userID")
}
```
Make a request
```gql
query MyQuery {
  todoMany(pagination: {first: 10}) {
    edges {
      node {
        id
        user {
          id
          name
        }
      }
    }
  }
}
```
## Expected Result
The query should return a todo list along with users
```json
{
  "data": {
    "todoMany": {
      "edges": [
        {
          "node": {
            "id": "01HQJJPZWVJDSVMKC38NYW7HK5",
            "user": {
              "id": "01HQJJPRE3HM0V0724EQR3S7YA",
              "name": "test user"
            }
          }
        }
      ]
    }
  }
}
```
## Actual Result
```json
{
  "errors": [
    {
      "message": "internal server error"
    }
  ],
  "data": null
}
```

## Environment
```env
FEDERATED_GRAPH_NAME=production
LISTEN_ADDR=0.0.0.0:3001
ROUTER_CONFIG_PATH=./config.json
METRICS_OTLP_ENABLED=false
PROMETHEUS_ENABLED=false
LOG_LEVEL=debug
DEV_MODE=true
```

### Router configuration

```yaml
version: "1"

cors:
  allow_origins: ["https://studio.apollographql.com", "http://gateway:4000", "https://localhost:4200"]
  allow_methods:
    - OPTIONS
    - GET
    - POST
  allow_headers:
    - Origin
    - Content-Length
    - Content-Type
  allow_credentials: true

telemetry:
  tracing:
    enabled: false
```

### Router execution config

```json
{"engineConfig":{"defaultFlushInterval":"500","datasourceConfigurations":[{"kind":"GRAPHQL","rootNodes":[{"typeName":"Query","fieldNames":["user","userMany"]},{"typeName":"Mutation","fieldNames":["createUser","updateUser","deleteUser"]},{"typeName":"Todo","fieldNames":["user","userID"]},{"typeName":"User","fieldNames":["id","name"]}],"childNodes":[{"typeName":"PageInfo","fieldNames":["endCursor","hasNextPage","hasPreviousPage","startCursor"]},{"typeName":"UserEdge","fieldNames":["cursor","node"]},{"typeName":"UserPagination","fieldNames":["edges","pageInfo","totalCount"]}],"overrideFieldPathFromAlias":true,"customGraphql":{"fetch":{"url":{"staticVariableContent":"http://localhost:8001/graphql"},"method":"POST","body":{},"baseUrl":{},"path":{}},"subscription":{"enabled":true,"url":{"staticVariableContent":"http://localhost:8001/graphql"},"protocol":"GRAPHQL_SUBSCRIPTION_PROTOCOL_WS"},"federation":{"enabled":true,"serviceSdl":"\nscalar Date\n\nenum Direction {\n  ascending\n  descending\n}\n\ntype PageInfo @shareable {\n  endCursor: String\n  hasNextPage: Boolean!\n  hasPreviousPage: Boolean!\n  startCursor: String\n}\n\nenum UserSort {\n    name\n}\n\ninput UserFilter {\n    \"\"\"Equal to name\"\"\"\n    nameEQ: String\n    \"\"\"Not equal to name\"\"\"\n    nameNEQ: String\n}\n\ninput PageArguments {\n  after: String\n  before: String\n  first: Int\n  last: Int\n}\n\ntype Todo @key(fields: \"userID\") {\n    userID: ID @external\n    user: User @requires(fields: \"userID\")\n}\n\ntype User @key(fields: \"id\") {\n    id: ID\n    name: String!\n}\n\ntype UserEdge {\n    cursor: String!\n    node: User!\n}\n\ntype UserPagination {\n    edges: [UserEdge!]!\n    pageInfo: PageInfo!\n    totalCount: Int!\n}\n\n\n\n\nextend type Query {\n    user(id: ID!): User!\n    userMany(\n        filtering: UserFilter\n        \"\"\"Sorting direction\"\"\"\n        direction: Direction\n        sortingFields: [UserSort]\n        \"\"\"List of fields by which you can search\"\"\"\n        pagination: PageArguments!\n    ): UserPagination!\n}\n\nextend type Mutation {\n    createUser(\n        name: String!\n    ): User!\n    updateUser(\n        id: ID!\n        name: String\n    ): User!\n    deleteUser(\n        \"\"\"Id user which we want to delete\"\"\"\n        id: ID!\n    ): Boolean!\n}\n"},"upstreamSchema":{"key":"22729f6159c5766d73d29ace2eb5338928e93554"}},"requestTimeoutSeconds":"10","id":"0","keys":[{"typeName":"Todo","selectionSet":"userID"},{"typeName":"User","selectionSet":"id"}],"requires":[{"typeName":"Todo","fieldName":"user","selectionSet":"userID"}]},{"kind":"GRAPHQL","rootNodes":[{"typeName":"Query","fieldNames":["todo","todoMany"]},{"typeName":"Mutation","fieldNames":["createTodo","updateTodo","deleteTodo"]},{"typeName":"Todo","fieldNames":["id","name","userID"]},{"typeName":"User","fieldNames":["todoMany","id"]}],"childNodes":[{"typeName":"PageInfo","fieldNames":["endCursor","hasNextPage","hasPreviousPage","startCursor"]},{"typeName":"TodoEdge","fieldNames":["cursor","node"]},{"typeName":"TodoPagination","fieldNames":["edges","pageInfo","totalCount"]}],"overrideFieldPathFromAlias":true,"customGraphql":{"fetch":{"url":{"staticVariableContent":"http://localhost:8002/graphql"},"method":"POST","body":{},"baseUrl":{},"path":{}},"subscription":{"enabled":true,"url":{"staticVariableContent":"http://localhost:8002/graphql"},"protocol":"GRAPHQL_SUBSCRIPTION_PROTOCOL_WS"},"federation":{"enabled":true,"serviceSdl":"\nscalar Date\n\nenum Direction {\n  ascending\n  descending\n}\n\ntype PageInfo @shareable {\n  endCursor: String\n  hasNextPage: Boolean!\n  hasPreviousPage: Boolean!\n  startCursor: String\n}\n\nenum TodoSort {\n    name\n}\n\ninput TodoFilter {\n    \"\"\"Equal to userID\"\"\"\n    userIDEQ: ID\n    \"\"\"Not equal to userID\"\"\"\n    userIDNEQ: ID\n}\n\ninput PageArguments {\n  after: String\n  before: String\n  first: Int\n  last: Int\n}\n\ntype Todo @key(fields: \"id, userID\") {\n    id: ID\n    name: String!\n    userID: ID!\n}\n\ntype User @key(fields: \"id\") {\n    id: ID @external\n    todoMany: [Todo] @requires(fields: \"id\")\n}\n\ntype TodoEdge {\n    cursor: String!\n    node: Todo!\n}\n\ntype TodoPagination {\n    edges: [TodoEdge!]!\n    pageInfo: PageInfo!\n    totalCount: Int!\n}\n\n\n\n\nextend type Query {\n    todo(id: ID!): Todo!\n    todoMany(\n        filtering: TodoFilter\n        \"\"\"Sorting direction\"\"\"\n        direction: Direction\n        sortingFields: [TodoSort]\n        \"\"\"List of fields by which you can search\"\"\"\n        pagination: PageArguments!\n    ): TodoPagination!\n}\n\nextend type Mutation {\n    createTodo(\n        name: String!\n        userID: ID!\n    ): Todo!\n    updateTodo(\n        id: ID!\n        name: String\n        userID: ID\n    ): Todo!\n    deleteTodo(\n        \"\"\"Id todo which we want to delete\"\"\"\n        id: ID!\n    ): Boolean!\n}\n"},"upstreamSchema":{"key":"1be40c74d22257537b418a7bbba875c74022beee"}},"requestTimeoutSeconds":"10","id":"1","keys":[{"typeName":"Todo","selectionSet":"id userID"},{"typeName":"User","selectionSet":"id"}],"requires":[{"typeName":"User","fieldName":"todoMany","selectionSet":"id"}]}],"fieldConfigurations":[{"typeName":"Query","fieldName":"user","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Query","fieldName":"userMany","argumentsConfiguration":[{"name":"filtering","sourceType":"FIELD_ARGUMENT"},{"name":"direction","sourceType":"FIELD_ARGUMENT"},{"name":"sortingFields","sourceType":"FIELD_ARGUMENT"},{"name":"pagination","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Query","fieldName":"todo","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Query","fieldName":"todoMany","argumentsConfiguration":[{"name":"filtering","sourceType":"FIELD_ARGUMENT"},{"name":"direction","sourceType":"FIELD_ARGUMENT"},{"name":"sortingFields","sourceType":"FIELD_ARGUMENT"},{"name":"pagination","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"createUser","argumentsConfiguration":[{"name":"name","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"updateUser","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"},{"name":"name","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"deleteUser","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"createTodo","argumentsConfiguration":[{"name":"name","sourceType":"FIELD_ARGUMENT"},{"name":"userID","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"updateTodo","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"},{"name":"name","sourceType":"FIELD_ARGUMENT"},{"name":"userID","sourceType":"FIELD_ARGUMENT"}]},{"typeName":"Mutation","fieldName":"deleteTodo","argumentsConfiguration":[{"name":"id","sourceType":"FIELD_ARGUMENT"}]}],"graphqlSchema":"directive @tag(name: String!) repeatable on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\ndirective @authenticated on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\ndirective @inaccessible on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\ndirective @requiresScopes(scopes: [[openfed__Scope!]!]!) on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\nscalar openfed__Scope\n\nscalar Date\n\nenum Direction {\n  ascending\n  descending\n}\n\ntype PageInfo {\n  endCursor: String\n  hasNextPage: Boolean!\n  hasPreviousPage: Boolean!\n  startCursor: String\n}\n\nenum UserSort {\n  name\n}\n\ninput UserFilter {\n  \"\"\"Equal to name\"\"\"\n  nameEQ: String\n\n  \"\"\"Not equal to name\"\"\"\n  nameNEQ: String\n}\n\ninput PageArguments {\n  after: String\n  before: String\n  first: Int\n  last: Int\n}\n\ntype Todo {\n  userID: ID\n  user: User\n  id: ID\n  name: String!\n}\n\ntype User {\n  id: ID\n  name: String!\n  todoMany: [Todo]\n}\n\ntype UserEdge {\n  cursor: String!\n  node: User!\n}\n\ntype UserPagination {\n  edges: [UserEdge!]!\n  pageInfo: PageInfo!\n  totalCount: Int!\n}\n\nenum TodoSort {\n  name\n}\n\ninput TodoFilter {\n  \"\"\"Equal to userID\"\"\"\n  userIDEQ: ID\n\n  \"\"\"Not equal to userID\"\"\"\n  userIDNEQ: ID\n}\n\ntype TodoEdge {\n  cursor: String!\n  node: Todo!\n}\n\ntype TodoPagination {\n  edges: [TodoEdge!]!\n  pageInfo: PageInfo!\n  totalCount: Int!\n}\n\ntype Query {\n  user(id: ID!): User!\n  userMany(\n    filtering: UserFilter\n\n    \"\"\"Sorting direction\"\"\"\n    direction: Direction\n    sortingFields: [UserSort]\n\n    \"\"\"List of fields by which you can search\"\"\"\n    pagination: PageArguments!\n  ): UserPagination!\n  todo(id: ID!): Todo!\n  todoMany(\n    filtering: TodoFilter\n\n    \"\"\"Sorting direction\"\"\"\n    direction: Direction\n    sortingFields: [TodoSort]\n\n    \"\"\"List of fields by which you can search\"\"\"\n    pagination: PageArguments!\n  ): TodoPagination!\n}\n\ntype Mutation {\n  createUser(name: String!): User!\n  updateUser(id: ID!, name: String): User!\n  deleteUser(\n    \"\"\"Id user which we want to delete\"\"\"\n    id: ID!\n  ): Boolean!\n  createTodo(name: String!, userID: ID!): Todo!\n  updateTodo(id: ID!, name: String, userID: ID): Todo!\n  deleteTodo(\n    \"\"\"Id todo which we want to delete\"\"\"\n    id: ID!\n  ): Boolean!\n}","stringStorage":{"22729f6159c5766d73d29ace2eb5338928e93554":"schema {\n  query: Query\n  mutation: Mutation\n}\n\ndirective @authenticated on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\ndirective @composeDirective(name: String!) repeatable on SCHEMA\n\ndirective @eventsPublish(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @eventsRequest(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @eventsSubscribe(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @extends on INTERFACE | OBJECT\n\ndirective @external on FIELD_DEFINITION | OBJECT\n\ndirective @inaccessible on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\ndirective @interfaceObject on OBJECT\n\ndirective @key(fields: openfed__FieldSet!, resolvable: Boolean = true) repeatable on INTERFACE | OBJECT\n\ndirective @link(as: String, for: String, import: [String], url: String!) repeatable on SCHEMA\n\ndirective @override(from: String!) on FIELD_DEFINITION\n\ndirective @provides(fields: openfed__FieldSet!) on FIELD_DEFINITION\n\ndirective @requires(fields: openfed__FieldSet!) on FIELD_DEFINITION\n\ndirective @requiresScopes(scopes: [[openfed__Scope!]!]!) on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\ndirective @shareable on FIELD_DEFINITION | OBJECT\n\ndirective @tag(name: String!) repeatable on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\nscalar Date\n\nenum Direction {\n  ascending\n  descending\n}\n\ntype Mutation {\n  createUser(name: String!): User!\n  deleteUser(\n    \"\"\"Id user which we want to delete\"\"\"\n    id: ID!\n  ): Boolean!\n  updateUser(id: ID!, name: String): User!\n}\n\ninput PageArguments {\n  after: String\n  before: String\n  first: Int\n  last: Int\n}\n\ntype PageInfo {\n  endCursor: String @shareable\n  hasNextPage: Boolean! @shareable\n  hasPreviousPage: Boolean! @shareable\n  startCursor: String @shareable\n}\n\ntype Query {\n  user(id: ID!): User!\n  userMany(\n    \"\"\"Sorting direction\"\"\"\n    direction: Direction\n    filtering: UserFilter\n    \"\"\"List of fields by which you can search\"\"\"\n    pagination: PageArguments!\n    sortingFields: [UserSort]\n  ): UserPagination!\n}\n\ntype Todo @key(fields: \"userID\") {\n  user: User @requires(fields: \"userID\")\n  userID: ID @external\n}\n\ntype User @key(fields: \"id\") {\n  id: ID\n  name: String!\n}\n\ntype UserEdge {\n  cursor: String!\n  node: User!\n}\n\ninput UserFilter {\n  \"\"\"Equal to name\"\"\"\n  nameEQ: String\n  \"\"\"Not equal to name\"\"\"\n  nameNEQ: String\n}\n\ntype UserPagination {\n  edges: [UserEdge!]!\n  pageInfo: PageInfo!\n  totalCount: Int!\n}\n\nenum UserSort {\n  name\n}\n\nscalar openfed__FieldSet\n\nscalar openfed__Scope","1be40c74d22257537b418a7bbba875c74022beee":"schema {\n  query: Query\n  mutation: Mutation\n}\n\ndirective @authenticated on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\ndirective @composeDirective(name: String!) repeatable on SCHEMA\n\ndirective @eventsPublish(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @eventsRequest(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @eventsSubscribe(sourceID: String, topic: String!) on FIELD_DEFINITION\n\ndirective @extends on INTERFACE | OBJECT\n\ndirective @external on FIELD_DEFINITION | OBJECT\n\ndirective @inaccessible on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\ndirective @interfaceObject on OBJECT\n\ndirective @key(fields: openfed__FieldSet!, resolvable: Boolean = true) repeatable on INTERFACE | OBJECT\n\ndirective @link(as: String, for: String, import: [String], url: String!) repeatable on SCHEMA\n\ndirective @override(from: String!) on FIELD_DEFINITION\n\ndirective @provides(fields: openfed__FieldSet!) on FIELD_DEFINITION\n\ndirective @requires(fields: openfed__FieldSet!) on FIELD_DEFINITION\n\ndirective @requiresScopes(scopes: [[openfed__Scope!]!]!) on ENUM | FIELD_DEFINITION | INTERFACE | OBJECT | SCALAR\n\ndirective @shareable on FIELD_DEFINITION | OBJECT\n\ndirective @tag(name: String!) repeatable on ARGUMENT_DEFINITION | ENUM | ENUM_VALUE | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | INPUT_OBJECT | INTERFACE | OBJECT | SCALAR | UNION\n\nscalar Date\n\nenum Direction {\n  ascending\n  descending\n}\n\ntype Mutation {\n  createTodo(name: String!, userID: ID!): Todo!\n  deleteTodo(\n    \"\"\"Id todo which we want to delete\"\"\"\n    id: ID!\n  ): Boolean!\n  updateTodo(id: ID!, name: String, userID: ID): Todo!\n}\n\ninput PageArguments {\n  after: String\n  before: String\n  first: Int\n  last: Int\n}\n\ntype PageInfo {\n  endCursor: String @shareable\n  hasNextPage: Boolean! @shareable\n  hasPreviousPage: Boolean! @shareable\n  startCursor: String @shareable\n}\n\ntype Query {\n  todo(id: ID!): Todo!\n  todoMany(\n    \"\"\"Sorting direction\"\"\"\n    direction: Direction\n    filtering: TodoFilter\n    \"\"\"List of fields by which you can search\"\"\"\n    pagination: PageArguments!\n    sortingFields: [TodoSort]\n  ): TodoPagination!\n}\n\ntype Todo @key(fields: \"id, userID\") {\n  id: ID\n  name: String!\n  userID: ID!\n}\n\ntype TodoEdge {\n  cursor: String!\n  node: Todo!\n}\n\ninput TodoFilter {\n  \"\"\"Equal to userID\"\"\"\n  userIDEQ: ID\n  \"\"\"Not equal to userID\"\"\"\n  userIDNEQ: ID\n}\n\ntype TodoPagination {\n  edges: [TodoEdge!]!\n  pageInfo: PageInfo!\n  totalCount: Int!\n}\n\nenum TodoSort {\n  name\n}\n\ntype User @key(fields: \"id\") {\n  id: ID @external\n  todoMany: [Todo] @requires(fields: \"id\")\n}\n\nscalar openfed__FieldSet\n\nscalar openfed__Scope"}},"subgraphs":[{"id":"0","name":"user","routingUrl":"http://localhost:8001/graphql"},{"id":"1","name":"todo","routingUrl":"http://localhost:8002/graphql"}]}
```

### Log output

```shell
12:06:40 PM INFO cmd/main.go:62 Found default config file. Values in the config file have higher priority than environment variables {"component": "@wundergraph/router", "service_version": "0.69.0", "config_file": "config.yaml"}
12:06:40 PM DEBUG maxprocs/maxprocs.go:47 maxprocs: Leaving GOMAXPROCS=10: CPU quota undefined {"component": "@wundergraph/router", "service_version": "0.69.0"}
12:06:40 PM WARN core/router.go:364 No graph token provided. The following features are disabled. Not recommended for Production. {"component": "@wundergraph/router", "service_version": "0.69.0", "features": ["Schema Usage Tracking", "Persistent operations"]}
12:06:40 PM WARN core/router.go:379 Development mode enabled. This should only be used for testing purposes {"component": "@wundergraph/router", "service_version": "0.69.0"}
12:06:40 PM INFO core/router.go:711 Static router config provided. Polling is disabled. Updating router config is only possible by providing a config. {"component": "@wundergraph/router", "service_version": "0.69.0"}
12:06:40 PM WARN core/router.go:949 Advanced Request Tracing (ART) is enabled in development mode but requires a graph token to work in production. For more information see https://cosmo-docs.wundergraph.com/router/advanced-request-tracing-art {"component": "@wundergraph/router", "service_version": "0.69.0"}
12:06:40 PM INFO core/router.go:983 Serving GraphQL playground {"component": "@wundergraph/router", "service_version": "0.69.0", "url": "http://0.0.0.0:3001/"}
12:06:40 PM INFO core/router.go:1030 Rate limiting disabled {"component": "@wundergraph/router", "service_version": "0.69.0"}
12:06:40 PM INFO core/router.go:1113 GraphQL endpoint {"component": "@wundergraph/router", "service_version": "0.69.0", "method": "POST", "url": "http://0.0.0.0:3001/graphql"}
12:06:40 PM INFO core/router.go:469 Server listening {"component": "@wundergraph/router", "service_version": "0.69.0", "listen_addr": "0.0.0.0:3001", "playground": true, "introspection": true, "config_version": ""}
12:06:46 PM ERROR core/graphql_prehandler.go:292 failed to plan operation {"component": "@wundergraph/router", "service_version": "0.69.0", "reqId": "MBP-Rost.fritz.box/cDly8hYaIk-000001", "error": "bad datasource configuration - could not plan the operation"}
github.com/wundergraph/cosmo/router/core.(*PreHandler).Handler-fm.(*PreHandler).Handler.func1
	github.com/wundergraph/cosmo/router/core/graphql_prehandler.go:292
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/core.(*WebsocketHandler).ServeHTTP
	github.com/wundergraph/cosmo/router/core/websocket.go:176
github.com/go-chi/chi/v5.(*Mux).ServeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:73
github.com/go-chi/chi/v5.(*Mux).Mount.func1
	github.com/go-chi/chi/v5@v5.0.11/mux.go:315
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5.(*Mux).routeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:443
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/pkg/cors.(*cors).ServeHTTP
	github.com/wundergraph/cosmo/router/pkg/cors/config.go:74
github.com/wundergraph/cosmo/router/internal/requestlogger.(*handler).ServeHTTP
	github.com/wundergraph/cosmo/router/internal/requestlogger/requestlogger.go:99
github.com/go-chi/chi/v5/middleware.RealIP.func1
	github.com/go-chi/chi/v5@v5.0.11/middleware/realip.go:36
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5/middleware.RequestID.func1
	github.com/go-chi/chi/v5@v5.0.11/middleware/request_id.go:76
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/internal/recoveryhandler.(*handler).ServeHTTP
	github.com/wundergraph/cosmo/router/internal/recoveryhandler/recovery.go:103
github.com/wundergraph/cosmo/router/core.(*Router).newServer.func2.1
	github.com/wundergraph/cosmo/router/core/router.go:836
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5.(*Mux).ServeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:90
net/http.serverHandler.ServeHTTP
	net/http/server.go:3137
net/http.(*conn).serve
	net/http/server.go:2039
12:06:46 PM ERROR core/graphql_handler.go:357 internal error {"component": "@wundergraph/router", "service_version": "0.69.0", "reqId": "MBP-Rost.fritz.box/cDly8hYaIk-000001", "error": "1 error occurred:\n\t* bad datasource configuration - could not plan the operation\n\n"}
github.com/wundergraph/cosmo/router/core.logInternalErrorsFromReport
	github.com/wundergraph/cosmo/router/core/graphql_handler.go:357
github.com/wundergraph/cosmo/router/core.(*PreHandler).writeOperationError
	github.com/wundergraph/cosmo/router/core/graphql_prehandler.go:418
github.com/wundergraph/cosmo/router/core.(*PreHandler).Handler-fm.(*PreHandler).Handler.func1
	github.com/wundergraph/cosmo/router/core/graphql_prehandler.go:293
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/core.(*WebsocketHandler).ServeHTTP
	github.com/wundergraph/cosmo/router/core/websocket.go:176
github.com/go-chi/chi/v5.(*Mux).ServeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:73
github.com/go-chi/chi/v5.(*Mux).Mount.func1
	github.com/go-chi/chi/v5@v5.0.11/mux.go:315
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5.(*Mux).routeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:443
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/pkg/cors.(*cors).ServeHTTP
	github.com/wundergraph/cosmo/router/pkg/cors/config.go:74
github.com/wundergraph/cosmo/router/internal/requestlogger.(*handler).ServeHTTP
	github.com/wundergraph/cosmo/router/internal/requestlogger/requestlogger.go:99
github.com/go-chi/chi/v5/middleware.RealIP.func1
	github.com/go-chi/chi/v5@v5.0.11/middleware/realip.go:36
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5/middleware.RequestID.func1
	github.com/go-chi/chi/v5@v5.0.11/middleware/request_id.go:76
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/wundergraph/cosmo/router/internal/recoveryhandler.(*handler).ServeHTTP
	github.com/wundergraph/cosmo/router/internal/recoveryhandler/recovery.go:103
github.com/wundergraph/cosmo/router/core.(*Router).newServer.func2.1
	github.com/wundergraph/cosmo/router/core/router.go:836
net/http.HandlerFunc.ServeHTTP
	net/http/server.go:2166
github.com/go-chi/chi/v5.(*Mux).ServeHTTP
	github.com/go-chi/chi/v5@v5.0.11/mux.go:90
net/http.serverHandler.ServeHTTP
	net/http/server.go:3137
net/http.(*conn).serve
	net/http/server.go:2039
12:06:46 PM INFO requestlogger/requestlogger.go:143 /graphql {"component": "@wundergraph/router", "service_version": "0.69.0", "status": 500, "method": "POST", "path": "/graphql", "query": "", "ip": "[REDACTED]", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.2.1 Safari/605.1.15", "latency": 0.017435916, "config_version": "", "request_id": "MBP-Rost.fritz.box/cDly8hYaIk-000001"}
```

**Comment by Aenimus** [comment]
Hi @ZirgVoice

Thank you for the report. I will investigate this at some point this week and update.

**Comment by zaichengwang** [comment]
Hi, any luck on this?
Thanks

**Comment by Aenimus** [comment]
Hi @zaichengwang @ZirgVoice

This should be fixed by composing with`wgc@0.51.1`+, and compatible with routers after `0.49.0`.

We recommend using the latest router available (currently `0.82.0`).

Thank you for the report and patience!
